test/cluster/test_tablets2.py

```diff
@@ -1835,3 +1835,39 @@ async def test_tablet_cleanup_vs_snapshot_race(manager: ManagerClient):
         await s0_log.wait_for('Cleanup failed for tablet', from_mark=s0_mark)
 
         await manager.api.take_snapshot(servers[0].ip_addr, ks, "test_snapshot")
+
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_truncate_during_topology_change(manager: ManagerClient):
+    """Test truncate operation during topology change."""
+
+    # Start 3 node cluster
+    cfg = {'enable_tablets': True}
+    servers = await manager.servers_add(3, config=cfg)
+    cql = manager.get_cql()
+    await cql.run_async("CREATE KEYSPACE ks WITH replication = {'class': 'NetworkTopologyStrategy', 'replication_factor': 3}")
+    await cql.run_async("CREATE TABLE ks.test (k int PRIMARY KEY, v int)")
+
+    logger.info("Populating table")
+    keys = range(10000)
+    stmt = cql.prepare("INSERT INTO ks.test (k, v) VALUES (?, ?)")
+    stmt.consistency_level = ConsistencyLevel.QUORUM
+    await asyncio.gather(*[cql.run_async(stmt, [k, k]) for k in keys])
+    await manager.api.keyspace_flush(servers[0].ip_addr, "ks", "test")
+
+    async def truncate_table():
+        await asyncio.sleep(10)
```

Comment:
wont this became flaky?

Reply:
we have `delay_bootstrap_120s`, so this 10 second shouldn't create flakiness.